### PR TITLE
improve handling of invalid IPPorts

### PR DESCRIPTION
**Pull request by josharian**

### CODE REVIEW

File: netaddr.go [hunk 1]
```diff
@@ -973,13 +973,20 @@ func MustParseIPPort(s string) IPPort {
 // IsZero reports whether p is its zero value.
 func (p IPPort) IsZero() bool { return p == IPPort{} }
 
+// Valid reports whether p.IP() is non-zero.
```
Comment (bradfitz) [inline netaddr.go:976]: Worth explicitly mentioning that the port is ignored, or that a zero port may be valid?

File: netaddr_test.go [hunk 2]
```diff
@@ -2726,3 +2726,19 @@ func TestNoAllocs(t *testing.T) {
...
 	})
 }
+
+func TestIPPortString(t *testing.T) {
+	tests := []struct {
+		ipp  IPPort
+		want string
+	}{
+		{IPPort{}, "invalid IPPort"},
+		{IPPortFrom(IP{}, 80), "invalid IPPort"},
```
Comment (bradfitz) [inline netaddr_test.go:2736]: add a test to lock in behavior of non-zero IP and zero port.

Comment (josharian) [reply netaddr_test.go:2736]: We have a fair amount of those in TestParseIPPort. I'll name this function to make it clearer that it is about invalid IPPorts.

File: netaddr.go (hunk 1 above)
Comment (josharian) [reply netaddr.go:976]: Done.

File: netaddr.go [hunk 3]
```diff
@@ -971,15 +971,23 @@ func MustParseIPPort(s string) IPPort {
 }
 
 // IsZero reports whether p is its zero value.
+// All ports are valid, including zero.
```
Comment (danderson) [inline netaddr.go:974]: This comment should go in Valid's docstring or the type's docstring, not IsZero's.

**Review by danderson — APPROVED** [review]

File: netaddr.go (hunk 3 above)
Comment (josharian) [reply netaddr.go]: Oops. Fixed, thanks.
